File: GymFlow-Backend/src/main/java/com/gymflow/entity/Coach.java
package com.gymflow.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;
import java.math.BigDecimal;
import java.time.LocalDateTime;

@Data
@TableName("coach")
public class Coach {

    @TableId(type = IdType.AUTO)
    private Long id;

    @TableField("real_name")
    private String realName;

    private String phone;

    private String password;

    // 0-女，1-男
    private Integer gender;

    private String specialty;

    private String certifications;

    @TableField("years_of_experience")
    private Integer yearsOfExperience;

    private Integer status;

    private BigDecimal rating;

    private String introduction;

    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createTime;

    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updateTime;
}

File: GymFlow-Web/src/types/coach.ts
// src/types/coach.ts

// 教练列表项
export interface CoachListVO {
  id: number
  realName: string
  phone: string
  gender?: number // 新增
  genderDesc?: string // 新增
  specialty?: string
  certifications?: string[]
  yearsOfExperience?: number
  status?: number
  statusDesc?: string
  totalCourses?: number
  // rating?: number
  createTime?: string
}

// 教练基础信息DTO
export interface CoachBasicDTO {
  id?: number
  realName: string
  phone: string
  password?: string
  gender: number // 新增，必填
  specialty?: string
  certificationList?: string[]
  yearsOfExperience?: number
  introduction?: string
}

// 教练详情
export interface CoachDetail {
  id: number
  realName: string
  phone: string
  gender: number // 新增
  genderDesc?: string // 新增
  specialty?: string
  certificationList?: string[]
  yearsOfExperience?: number
  status?: number
  statusDesc?: string
  totalCourses?: number
  // rating?: number
  introduction?: string
  createTime?: string
  updateTime?: string
  schedules?: CoachScheduleList[]
  courses?: CoachCourseList[]
}